addons/account/tests/test_transfer_wizard.py

```diff
@@ -289,3 +291,49 @@ def test_transfer_wizard_no_currency_conversion(self):
         self.assertEqual(len(destination_lines), 2, "Two lines should have been created on destination account: one for each currency (the lines with same partner and currency should have been aggregated)")
         self.assertAlmostEqual(destination_lines.filtered(lambda x: x.currency_id == self.test_currency_1).amount_currency, -10, self.test_currency_1.decimal_places)
         self.assertAlmostEqual(destination_lines.filtered(lambda x: x.currency_id == self.test_currency_2).amount_currency, -756, self.test_currency_2.decimal_places)
+
+    def test_period_change_lock_date(self):
+        """ Test that the period change wizard correctly handles the lock date: if the original entry is dated
+        before the lock date, the adjustment entry is created on the first end of month after the lock date.
+        """
+        # Set up accrual accounts
+        self.company_data['company'].expense_accrual_account_id = self.env['account.account'].create({
+            'name': 'Expense Accrual Account',
+            'code': '113226',
+            'account_type': 'asset_prepayments',
+            'reconcile': True,
+        })
+        self.company_data['company'].revenue_accrual_account_id = self.env['account.account'].create({
+            'name': 'Revenue Accrual Account',
+            'code': '226113',
+            'account_type': 'liability_current',
+            'reconcile': True,
+        })
+
+        # Create a move before the lock date
+        move = self.env['account.move'].create({
+            'journal_id': self.company_data['default_journal_sale'].id,
+            'date': '2019-01-01',
+            'line_ids': [
+                Command.create({'account_id': self.accounts[0].id, 'debit': 1000, }),
+                Command.create({'account_id': self.accounts[0].id, 'credit': 1000, }),
+            ]
+        })
+        move.action_post()
+
+        # Set the lock date
+        move.company_id.write({'period_lock_date': '2019-02-28', 'fiscalyear_lock_date': '2019-02-28'})
+
+        # Use the transfer wizard to change the entry date
+        wizard = self.env['account.automatic.entry.wizard'] \
+            .with_context(active_model='account.move.line', active_ids=move.line_ids[0].ids) \
+            .create({
+                'action': 'change_period',
+                'date': '2019-05-01',
+                'journal_id': self.company_data['default_journal_misc'].id,
+            })
+        wizard_res = wizard.do_action()
```

Comment:
Maybe just check that the date is displayed in the wizard, as there is no warning whatsoever anywhere else that it won't be on the date of the original move